orchagent/mirrororch.cpp

```diff
@@ -562,175 +601,165 @@ bool MirrorOrch::updateSessionDstPort(const string& name, MirrorEntry& session)
-        {
-            continue;
-        }
-
-        // It is possible to have few sessions that points to one next hop
-        if (sessionIter->second.nexthopInfo.nexthop != update.entry.ip_address)
-        {
-            continue;
-        }
+        const auto& name = it->first;
+        auto& session = it->second;
 
-        SWSS_LOG_INFO("Updating neighbor info %s %s\n", update.entry.ip_address.to_string().c_str(), update.entry.alias.c_str());
+        SWSS_LOG_NOTICE("neighbor %s : %s", update.entry.ip_address.to_string().c_str(), update.entry.alias.c_str());
 
-        const auto& name = sessionIter->first;
-        auto& session = sessionIter->second;
 
-        if (update.add)
+        // XXX: add comments!
+        if (session.dstIp != update.entry.ip_address &&
+                session.nexthopInfo.nexthop != update.entry.ip_address)
         {
-            if (!getNeighborInfo(name, session, update.entry, update.mac))
-            {
-                if (session.status)
-                {
-                    deactivateSession(name, session);
-                }
-               continue;
-            }
+            continue;
+        }
 
-            if (session.status)
-            {
-                if (!updateSessionDstMac(name, session))
-                {
-                    continue;
-                }
+        SWSS_LOG_NOTICE("Updating mirror session %s with neighbor %s",
+                name.c_str(), update.entry.alias.c_str());
 
-                if (!updateSessionDstPort(name, session))
-                {
-                    continue;
-                }
-            }
-            else
-            {
-                activateSession(name, session);
-            }
-        }
-        else if (session.status)
-        {
-            deactivateSession(name, session);
-            session.neighborInfo.resolved = false;
-        }
+        updateSession(name, session);
     }
 }
 
 void MirrorOrch::updateFdb(const FdbUpdate& update)
 {
     SWSS_LOG_ENTER();
 
-    auto sessionIter = m_syncdMirrors.begin();
-    for (; sessionIter != m_syncdMirrors.end(); ++sessionIter)
+    for (auto it = m_syncdMirrors.begin(); it != m_syncdMirrors.end(); it++)
     {
-        if (!sessionIter->second.neighborInfo.resolved ||
-                sessionIter->second.neighborInfo.port.m_type != Port::VLAN)
-        {
-            continue;
-        }
+        const auto& name = it->first;
+        auto& session = it->second;
 
-        // It is possible to have few session that points to one FDB entry
-        if (sessionIter->second.neighborInfo.mac != update.entry.mac ||
-                sessionIter->second.neighborInfo.vlanOid != update.entry.bv_id)
+        // Check the following three conditions:
+        // 1) mirror session is pointing to a VLAN
+        // 2) the VLAN matches the FDB notification VLAN ID
+        // 3) the destination MAC matches the FDB notifaction MAC
+        if (session.neighborInfo.port.m_type != Port::VLAN ||
```

Comment:
>session.neighborInfo.port.m_type != Port::VLAN || [](start = 12, length = 49)

the condition is wrong here, even for non-vlan case, you need to de-activate the session.

Reply:
will there be a non-vlan case?